google/cloud/internal/detect_gcp_platform.cc

```diff
@@ -0,0 +1,91 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/internal/detect_gcp_platform.h"
+#include "google/cloud/internal/filesystem.h"
+#include <fstream>
+#include <iostream>
+#include <regex>
+#ifdef __unix__
+#include <dirent.h>
+#include <sys/stat.h>
+#endif
+
+namespace google {
+namespace cloud {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace internal {
+
+bool HasGceBiosInfo(std::string const& path) {
+#ifdef _WIN32
+  return false;
```

Comment:
Should we care / handle windows? If so, we need to read from the registry. Do we run tests against windows?